FetchMigration/python/fetch_orchestrator.py

```diff
@@ -2,33 +2,100 @@
 import base64
 import logging
 import os
+import re
 import subprocess
 import sys
 from typing import Optional
 
+import yaml
+
 import metadata_migration
 import migration_monitor
+from fetch_orchestrator_params import FetchOrchestratorParams
 from metadata_migration_params import MetadataMigrationParams
 from migration_monitor_params import MigrationMonitorParams
 
+__PROTOCOL_PREFIX_PATTERN = re.compile(r"^https?://")
+__HTTPS_PREFIX = "https://"
 __DP_EXECUTABLE_SUFFIX = "/bin/data-prepper"
 __PIPELINE_OUTPUT_FILE_SUFFIX = "/pipelines/pipeline.yaml"
 
 
-def run(dp_base_path: str, dp_config_file: str, dp_endpoint: str) -> Optional[int]:
-    dp_exec_path = dp_base_path + __DP_EXECUTABLE_SUFFIX
-    output_file = dp_base_path + __PIPELINE_OUTPUT_FILE_SUFFIX
-    metadata_migration_params = MetadataMigrationParams(dp_config_file, output_file, report=True)
-    logging.info("Running pre-migration steps...\n")
+def __get_env_string(name: str) -> Optional[str]:
+    val: str = os.environ.get(name, "")
+    if len(val) > 0:
+        return val
+    else:
+        return None
+
+
+def update_target_host(dp_config: dict, target_host: str):
```

Comment:
Just to confirm this support urls like: `https:vpc-123:443` and `vpc-123`. I have an upcoming change that plans to store a full url in SSM versus the shortened form

Reply:
`https://vpc-123:443` would be left as-is (since it's already using HTTPS) but `vpc-123` would be changed to `https://vpc-123` to force HTTPS. Would that be an issue for your change?

Reply:
The first option was supposed to be `https://vpc-123:443`

Reply:
Sounds good wanted to ensure we could support both. We use the second currently but will be moving to the first.